### PR TITLE
updates autofill

### PR DESCRIPTION
**Pull request by brindy**
Task/Issue URL: https://app.asana.com/0/inbox/392891325557408/1200377512689983/1200407253760291
Tech Design URL:
CC:

**Description**:

Updates autofill to more securely message between JS and Swift layers.

**Steps to test this PR**:
1. Use each of the clients to test this commit.

**OS Testing**:

* [x] iOS 13
* [x] iOS 14
* [ ] macOS 10.15
* [x] macOS 11

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)

### CODE REVIEW

File: Tests/BrowserServicesKitTests/Autofill/AutofillEncrypterTests.swift [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+import Foundation
+import XCTest
+import CryptoKit
+@testable import BrowserServicesKit
+
+class AutofillEncrypterTests: XCTestCase {
+
+    func testWhenMessageIsEncrypted_ThenCanBeDecryptedUsingAuthenticationData() throws {
+        let key: [UInt8] = SymmetricKey(size: .bits256).withUnsafeBytes { $0.map { $0 }}
+        let iv: [UInt8] = SymmetricKey(size: .bits256).withUnsafeBytes { $0.map { $0 }}
```
Comment (samsymons) [inline Tests/BrowserServicesKitTests/Autofill/AutofillEncrypterTests.swift:29]: You could also do this here, purely a matter of personal preference though 🙂 

```suggestion
        let key: [UInt8] = SymmetricKey(size: .bits256).withUnsafeBytes { Array($0) }
        let iv: [UInt8] = SymmetricKey(size: .bits256).withUnsafeBytes { Array($0) }
```

**Comment by samsymons** [comment]
I'm seeing some new warnings for unused files, unclear how important these are to handle though:

```
found 6 file(s) which are unhandled; explicitly declare them as resources or exclude from the target
    /Users/samsymons/Code/DuckDuckGo/BrowserServicesKit/Sources/BrowserServicesKit/Resources/duckduckgo-autofill/dist/autofill-host-styles_firefox.css
    /Users/samsymons/Code/DuckDuckGo/BrowserServicesKit/Sources/BrowserServicesKit/Resources/duckduckgo-autofill/dist/autofill.css
    /Users/samsymons/Code/DuckDuckGo/BrowserServicesKit/Sources/BrowserServicesKit/Resources/duckduckgo-autofill/jest.setup.js
    /Users/samsymons/Code/DuckDuckGo/BrowserServicesKit/Sources/BrowserServicesKit/Resources/duckduckgo-autofill/dist/autofill-host-styles_chrome.css
    /Users/samsymons/Code/DuckDuckGo/BrowserServicesKit/Sources/BrowserServicesKit/Resources/duckduckgo-autofill/jest.config.js
    /Users/samsymons/Code/DuckDuckGo/BrowserServicesKit/Sources/BrowserServicesKit/Resources/duckduckgo-autofill/jest-test-environment.js
found 1 file(s) which are unhandled; explicitly declare them as resources or exclude from the target
    /Users/samsymons/Code/DuckDuckGo/BrowserServicesKit/Sources/SecureVault/Resources/vault-support.js
```

**Review by samsymons — APPROVED** [review]
Functionality-wise this looks good - there's a comment about unused file warnings that could be nice to clean up, otherwise this is solid. Great work!

File: Sources/SecureVault/Resources/vault-support.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
+//
+//  vault-support.js
```
[the remark this answers is not in the document]
Comment (brindy) [reply Sources/SecureVault/Resources/vault-support.js]: yeah there's a bunch of premature files here which I've deleted now

**Pull request merged**